api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -543,6 +565,148 @@ public int hashCode() {
     }
   }
 
+  public static class GeometryType extends PrimitiveType {
+
+    private final String crs;
+
+    private GeometryType(String crs) {
+      if (crs != null) {
+        Preconditions.checkArgument(!crs.isEmpty(), "Invalid CRS: (empty string)");
+        Preconditions.checkArgument(
+            crs.trim().equals(crs), "CRS must not have leading or trailing spaces: '%s'", crs);
+        this.crs = crs;
+      } else {
+        this.crs = null;
+      }
+    }
+
+    private GeometryType() {
+      crs = null;
+    }
+
+    public static GeometryType crs84() {
```

Comment:
Static factory methods typically go above constructors, and the no-arg constructor usually comes first.

Reply:
Reordered.

* Static members comes first
* Then static factory methods
* Then private members
* Then no-arg constructor
* Then other constructors